src/main/scala/mesosphere/marathon/api/v2/AppsResource.scala

```diff
@@ -60,9 +60,20 @@ class AppsResource @Inject() (
              @DefaultValue("false")@QueryParam("force") force: Boolean): Response = {
 
     val now = clock.now()
+
+    val bodyJson = Json.parse(body)
+    if (bodyJson.toString() == "[]")
```

Comment:
That seems to be a very special purpose check. You want to check if the user specified an empty array?